src/Driver/SolverTypes/MISSolverType.jl

```diff
@@ -0,0 +1,155 @@
+
+export MISSolverType
+
+"""
+# Description
+    MISSolverType(;
+        splitting_type = SlowFastSplitting(),
+        linear_model = AtmosAcousticGravityLinearModel,
+        mis_method = MIS2,
+        fast_method = LSRK54CarpenterKennedy,
+        nsubsteps = 50,
+    )
+
+This solver type constructs an ODE solver using a generalization
+of the split-explicit Runge-Kutta method. Known as the Multirate
+Infinitesimal Step (MIS) method, this solver solves ODEs with
+the partitioned form:
+
+```math
+    \\dot{Q} = f_fast(Q, t) + f_slow(Q, t)
+```
+
+where the right-hand-side functions `f_fast` and `f_slow` denote
+fast and slow dynamics respectively, depending on the state `Q`.
+
+# Arguments
+- `splitting_type` (DiscreteSplittingType): The type of discrete
+    splitting to apply to the right-hand side.
+    Default: `SlowFastSplitting()`
+- `linear_model` (Type): The linear model describing fast dynamics.
+    Default: `AtmosAcousticGravityLinearModel`
+- `mis_method` (Function): Function defining the particular MIS
+    method to be used.
+    Default: `MIS2`
+- `fast_method` (Function): Function defining the fast solver.
+    Default: `LSRK54CarpenterKennedy`
+- `nsubsteps` (Int): Integer denoting the total number of times
+    to substep the fast process.
+    Default: `50`
+
+### References
+    @article{KnothWensch2014,
+        title={Generalized split-explicit Runge--Kutta methods for
+            the compressible Euler equations},
+        author={Knoth, Oswald and Wensch, Joerg},
+        journal={Monthly Weather Review},
+        volume={142},
+        number={5},
+        pages={2067--2081},
+        year={2014}
+    }
+"""
+struct MISSolverType{DS} <: AbstractSolverType
+    # The type of discrete splitting to apply to the right-hand side
+    splitting_type::DS
+    # Linear model describing fast dynamics
+    linear_model::Type
```

Comment:
`fast_model` ? Doesn't have to be linear. Similarly for other multi-rate solvers.

Reply:
Really good point here, in fact, when we go full HEVI, `fast_model` will most certainly be non-linear.